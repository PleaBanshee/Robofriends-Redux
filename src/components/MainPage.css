@font-face {
    font-display: swap; /* Gives the font face an extremely small block period and an infinite swap period. */
    font-family: 'SEGA LOGO FONT';
    font-style: normal;
    font-weight: normal;
    src: local('SEGA LOGO FONT'), url('../containers/SEGA.woff') format('woff');
}
  
h1 {
    font-family: 'SEGA LOGO FONT';
    font-weight: 200;
    color: #0ccac4;
}