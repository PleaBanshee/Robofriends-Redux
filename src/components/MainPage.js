// Contains all elements on the Main Page
import React, { Component } from 'react';

import CardList from '../components/CardList';
import SearchBox from '../components/SearchBox';
import Scroll from '../components/Scroll';
import ErrorBoundary from '../components/ErrorBoundary.js';
import Header from '../components/Header';

class MainPage extends Component {
  // no constructor needed, because there is no more direct state, state is returned as props
  componentDidMount() {
    this.props.onRequestRobots(); // props returned
  }

  filterRobots = () => {
      const { robots, searchField } = this.props;
      return robots.filter(robot => {
      return robot.name.toLowerCase().includes(searchField.toLowerCase()); // check if search value exists in robots array. Works for upper and lowercase searches
  });
  }

  render() {
    const { onSearchChange, isPending } = this.props; // state receives props
    return (
      <div className='tc'>
      <Header/>
      <SearchBox searchChange={onSearchChange}/>
      <Scroll>
        { isPending ? <h1>Loading</h1> :
          !this.filterRobots().length? <h1>ROBOT DOESN'T EXIST...</h1>:
            <ErrorBoundary>
              <CardList robots={this.filterRobots()} />
            </ErrorBoundary>
        }
      </Scroll>
      </div>
    )}
}

export default MainPage;