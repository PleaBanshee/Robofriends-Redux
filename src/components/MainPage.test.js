import { shallow } from 'enzyme';
import MainPage from './MainPage';
  
let wrapper;
// runs before each test
beforeEach(() => {
    const mockProps = {
        onRequestRobots: jest.fn(), // mocks a function
        robots: [],
        searchField: '',
        isPending: false
    }
    // receives props from App.js
    wrapper = shallow(<MainPage {...mockProps}/>)
});

it('Renders Main Page successfully',() => {
    expect.assertions(1);
    expect(wrapper).toMatchSnapshot();
    // expect(wrapper.instance().filterRobots()).toEqual([]);
});

it('Robots filtered successfully',() => {
    expect.assertions(1);
    const mockProps2 = {
        onRequestRobots: jest.fn(),
        robots: [{
            id: 3,
            name: 'John',
            email: 'john@gmail.com'
        }],
        searchField: 'john',
        isPending: false
    }
    const wrapper2 = shallow(<MainPage {...mockProps2}/>)
    // instance(): provides access to all methods of the wrapper object
    expect(wrapper2.instance().filterRobots()).toEqual([{
        id: 3,
        name: 'John',
        email: 'john@gmail.com'
    }]);
});