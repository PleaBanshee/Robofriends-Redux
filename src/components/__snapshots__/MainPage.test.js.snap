// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Main Page successfully 1`] = `
<div
  className="tc"
>
  <Memo(Header) />
  <SearchBox />
  <Scroll>
    <h1>
      ROBOT DOESN'T EXIST...
    </h1>
  </Scroll>
</div>
`;
